lib/features/academic_calendar/widgets/academic_calendar_consumer.dart

```diff
@@ -0,0 +1,45 @@
+import "package:flutter/material.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+
+import "../../../config/ui_config.dart";
+import "../../../widgets/my_error_widget.dart";
+import "../../home_view/widgets/loading_widgets/horizontal_rectangular_section_loading.dart";
+import "../repository/academic_calendar_repo.dart";
+import "countdown_widget/exam_session_countdown.dart";
+import "home_screen_greeting.dart";
+
+class AcademicCalendarConsumer extends ConsumerWidget {
```

Comment:
I like naming convention here. Should all consumer widgets be named that way in other part of the project? Even private ones?

Reply:
I mean, I'm not enforcing it. I was just natural in this case. But yeah, if you want we can introduce some naming convention like this. But not to all `ConsumerWidget`s but to those that actually consumer some state - why not. Cause sometimes we also extend `ConsumerWidget` just to get `ref` for something